sign/ed25519/signapi.go

```diff
@@ -0,0 +1,75 @@
+package ed25519
+
+import (
+	"crypto/rand"
+	"encoding/asn1"
+
+	"github.com/cloudflare/circl/sign"
+)
+
+// Scheme is
+const Scheme = scheme(sign.Ed25519)
+
+type scheme sign.SchemeID
+
+func (scheme) ID() sign.SchemeID   { return sign.SchemeID(Scheme) }
+func (scheme) Name() string        { return "Ed25519" }
+func (scheme) PublicKeySize() int  { return PublicKeySize }
+func (scheme) PrivateKeySize() int { return PrivateKeySize }
+func (scheme) SignatureSize() int  { return SignatureSize }
+func (scheme) SeedSize() int       { return SeedSize }
+func (scheme) TLSIdentifier() uint { return 0x0807 }
+func (scheme) Oid() asn1.ObjectIdentifier {
+	return asn1.ObjectIdentifier{1, 3, 101, 112}
+}
+
+func (scheme) GenerateKey() (sign.PublicKey, sign.PrivateKey, error) {
+	return GenerateKey(rand.Reader)
+}
+
+func (scheme) Sign(
+	sk sign.PrivateKey,
+	message []byte,
+	opts *sign.SignatureOpts) []byte {
```

Comment:
nit: my preference would be to use a trailing comma in multi-line signatures.

Reply:
Yes, that's nicer.